ftplugin/markdown.vim

```diff
@@ -105,8 +105,17 @@ endfunction
 function! s:GetHeadingLinkGFM(headingName)
     let l:headingLink = tolower(a:headingName)
 
+    " \%^ : beginning of file
```

Comment:
`\%^` means beginning of string, and `\%$` means end of string. Test case for this:

```
call ASSERT(GetHeadingLinkTest("##### _Hello_World_", "GFM") ==# "hello_world")
```

I'll merge this PR and fix the comments later, thanks very much!